telemetry_test.go

```diff
@@ -14,332 +14,255 @@ import (
+			Message: map[string]string{
+				typeKey:    "client_telemetry_type",
+				queryIDKey: "123",
+			},
+			Timestamp: time.Now().UnixNano() / int64(time.Millisecond),
+		}); err != nil {
+			t.Fatal(err)
+		}
+		err = st.sendBatch()
+		if err == nil {
+			t.Fatal("should have failed")
+		}
+		if st.enabled == true {
+			t.Fatal("telemetry should be disabled")
+		}
+	})
 }
 
 func TestTelemetryDisabled(t *testing.T) {
-	config, err := ParseDSN(dsn)
-	if err != nil {
-		t.Error(err)
-	}
-	sc, err := buildSnowflakeConn(context.Background(), *config)
-	if err != nil {
-		t.Fatal(err)
-	}
-	if err = authenticateWithConfig(sc); err != nil {
-		t.Fatal(err)
-	}
-	sr := &snowflakeRestful{
-		FuncPost:      postTestAppBadGatewayError,
-		TokenAccessor: getSimpleTokenAccessor(),
-	}
-	st := &snowflakeTelemetry{
-		sr:        sr,
-		mutex:     &sync.Mutex{},
-		enabled:   false, // disable
-		flushSize: defaultFlushSize,
-	}
-	if err = st.addLog(&telemetryData{
-		Message: map[string]string{
-			typeKey:    "client_telemetry_type",
-			queryIDKey: "123",
-		},
-		Timestamp: time.Now().UnixNano() / int64(time.Millisecond),
-	}); err == nil {
-		t.Fatal("should have failed")
-	}
-	st.enabled = true
-	if err = st.addLog(&telemetryData{
-		Message: map[string]string{
-			typeKey:    "client_telemetry_type",
-			queryIDKey: "123",
-		},
-		Timestamp: time.Now().UnixNano() / int64(time.Millisecond),
-	}); err != nil {
-		t.Fatal(err)
-	}
-	st.enabled = false
-	err = st.sendBatch()
-	if err == nil {
-		t.Fatal("should have failed")
-	}
+	runSnowflakeConnTest(t, func(sct *SCTest) {
+		sr := &snowflakeRestful{
+			FuncPost:      postTestAppBadGatewayError,
+			TokenAccessor: getSimpleTokenAccessor(),
+		}
+		st := &snowflakeTelemetry{
+			sr:        sr,
+			mutex:     &sync.Mutex{},
+			enabled:   false, // disable
+			flushSize: defaultFlushSize,
+		}
+		if err := st.addLog(&telemetryData{
+			Message: map[string]string{
+				typeKey:    "client_telemetry_type",
+				queryIDKey: "123",
+			},
+			Timestamp: time.Now().UnixNano() / int64(time.Millisecond),
+		}); err == nil {
+			t.Fatal("should have failed")
+		}
+		st.enabled = true
+		if err := st.addLog(&telemetryData{
+			Message: map[string]string{
+				typeKey:    "client_telemetry_type",
+				queryIDKey: "123",
+			},
+			Timestamp: time.Now().UnixNano() / int64(time.Millisecond),
+		}); err != nil {
+			t.Fatal(err)
+		}
+		st.enabled = false
+		err := st.sendBatch()
+		if err == nil {
+			t.Fatal("should have failed")
+		}
+	})
 }
 
 func TestAddLogError(t *testing.T) {
-	config, err := ParseDSN(dsn)
-	if err != nil {
-		t.Error(err)
-	}
-	sc, err := buildSnowflakeConn(context.Background(), *config)
-	if err != nil {
-		t.Fatal(err)
-	}
-	if err = authenticateWithConfig(sc); err != nil {
-		t.Fatal(err)
-	}
-
-	sr := &snowflakeRestful{
-		FuncPost:      funcPostTelemetryRespFail,
-		TokenAccessor: getSimpleTokenAccessor(),
-	}
+	runSnowflakeConnTest(t, func(sct *SCTest) {
+		sr := &snowflakeRestful{
```

Comment:
Inline?

Reply:
Better not. It is easier to read in separation